aot-core/src/testFixtures/groovy/io/micronaut/aot/core/codegen/AbstractSourceGeneratorSpec.groovy

```diff
@@ -66,29 +66,31 @@ abstract class AbstractSourceGeneratorSpec extends Specification {
     void generate() {
         def sourceGenerator = newGenerator()
         sourceGenerator.generate(context)
-        def sources = context.getGeneratedJavaFiles().collectEntries([:]) {
+        def javaFiles = context.getGeneratedJavaFiles()
+        def sources = new HashMap<JavaFile, String>()
+        javaFiles.forEach {
             def writer = new StringWriter()
             it.writeTo(writer)
-            [it, writer.toString()]
+            sources.put(it, writer.toString())
         }
 
         this.generatedSources = new GeneratedSources(sources, context.getGeneratedStaticInitializers(), resourcesDir.toFile())
     }
 
     void assertThatGeneratedSources(@DelegatesTo(value=GeneratedSources, strategy = Closure.DELEGATE_FIRST) Closure<?> spec) {
         if (generatedSources == null) {
-            throw new AssertionError("Couldn't find generated sources. Did you call 'generateSources()'?")
+            throw new AssertionError((Object) "Couldn't find generated sources. Did you call 'generateSources()'?")
         }
         spec.delegate = generatedSources
         spec.resolveStrategy = Closure.DELEGATE_FIRST
         spec()
         def verified = generatedSources.verifiedClasses
         def missing = generatedSources.generatedClasses - verified
         if (missing) {
-            throw new AssertionError("Too few assertions. You also need to verify classes: $missing")
+            throw new AssertionError((Object) "Too few assertions. You also need to verify classes: $missing")
```

Comment:
suggestion: remove those nasty casts to `Object`, to keep readability and to reduce the diff size of this fix

Reply:
reverted